contracts/PLCR.sol

```diff
@@ -0,0 +1,398 @@
+pragma solidity 0.4.18;
+
+import "@aragon/os/contracts/apps/AragonApp.sol";
+import "@aragon/os/contracts/lib/misc/Migrations.sol";
+import "@aragon/os/contracts/lib/zeppelin/math/SafeMath.sol";
+import "@aragon/os/contracts/lib/zeppelin/math/SafeMath64.sol";
+
+import "@aragon/apps-staking/contracts/interfaces/IStaking.sol"; // TODO: unifiy with Curation somewhere
+import "@aragon/apps-curation/contracts/interfaces/IVoting.sol"; // TODO: unifiy with Curation somewhere
+
+
+contract PLCR is AragonApp, IVoting {
+    using SafeMath for uint256;
+    using SafeMath64 for uint64;
+
+    IStaking public staking;
+    uint256 public voteQuorum;
+    uint256 public minorityBlocSlash;
+    uint64 public commitDuration;
+    uint64 public revealDuration;
+
+    uint256 constant public PCT_BASE = 10 ** 18; // 0% = 0; 1% = 10^16; 100% = 10^18
+
+    bytes32 constant public CREATE_VOTE_ROLE = keccak256("CREATE_VOTE_ROLE");
+
+    // TODO!!!
+    enum TimeUnit { Blocks, Seconds }
+
+    struct UserVote {
+        uint256 lockId;
+        uint256 stake;
+        bytes32 secretHash;
+        bool revealed;
+        bool voteOption;
+        bool claimed;
+    }
+
+    struct Vote {
+        uint64 commitEndDate;
+        uint64 revealEndDate;
+        bool result;
+        bool computed;
+        string metadata;
```

Comment:
Thoughts on splitting `metadata` into it's own separate struct? My concern is that the field currently enables a wide spectrum of possible input. A metadata struct would enable a little more structure. 

```
    struct Metainformation {
        string title;
        string description;
    }
```
This is assuming we can align on fields, if not.. I don't think it's necessary